src/test/java/uk/gov/justice/digital/service/DomainDefinitionClientServiceTest.java

```diff
@@ -0,0 +1,153 @@
+package uk.gov.justice.digital.service;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.model.*;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import uk.gov.justice.digital.config.JobParameters;
+import uk.gov.justice.digital.domain.DomainExecutor;
+import uk.gov.justice.digital.domain.model.DomainDefinition;
+import uk.gov.justice.digital.service.DomainDefinitionClientService;
+import uk.gov.justice.digital.service.DomainService;
+
+import java.io.IOException;
+import java.util.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+
+class DomainDefinitionClientServiceTest {
+
+
+    private static DomainDefinitionClientService domainDefinitionService;
+    private static DomainService domainService;
+    private static AmazonDynamoDB dynamoDB;
+    private static ObjectMapper mapper;
+
+
+    @BeforeEach
+    void setUp() {
+        dynamoDB = mock(AmazonDynamoDB.class);
+        mapper = mock(ObjectMapper.class);
+        domainDefinitionService = new DomainDefinitionClientService(dynamoDB, mapper);
+        domainService = new DomainService(mock(JobParameters.class), domainDefinitionService,
+                mock(DomainExecutor.class));
+    }
+
+    @Test
+    void shouldExecuteQueryForGivenDomain() {
+        String domainTableName = "test";
+        String domainName = "incident";
+        QueryResult result = mock(QueryResult.class);
+        when(dynamoDB.query(any(QueryRequest.class))).thenReturn(result);
+        QueryResult expected_result = domainDefinitionService.executeQuery(domainTableName, domainName);
+        assert(expected_result.getItems().isEmpty());
+    }
+
+    @Test
+    void shouldReturnErrorIfDomainTableDoesntExists() {
+        String domainTableName = "test";
+        String domainName = "incident";
+        when(dynamoDB.query(any(QueryRequest.class))).thenThrow(ResourceNotFoundException.class);
+        try {
```

Comment:
Do not handle exceptions like this in your tests. If you need to test an exception is thrown then use `assertThrown`